### PR TITLE
Weston is added to the build dependencies, and is now run by xvfb.py.

### PR DESCRIPTION
**Pull request by alex-voodoo**
Browser tests typically require a window manager.  The one may not exist on test bots that do not provide graphics mode at all, and a substitute in form of openbox and xcompmgr running as a virtual WM is provided by means of test harness in testing/xvfb.py.  This was not enough, however, for the new graphics composing modes like Ozone upon Wayland because these need a Wayland server.

This change introduces launching weston (which provides a Wayland server) to allow tests to run with Wayland bakend.

### CODE REVIEW

File: testing/xvfb.py [hunk 1]
```diff
@@ -75,13 +75,19 @@ def run_executable(cmd, env, stdoutfile=None):
         xcompmgr_proc = subprocess.Popen('xcompmgr', stdout=subprocess.PIPE,
                                          stderr=subprocess.STDOUT, env=env)
 
+        # Tests may need a Wayland server.
+        weston_proc = subprocess.Popen(('weston', '--backend=x11-backend.so'),
+                                       stdout=subprocess.PIPE,
+                                       stderr=subprocess.STDOUT, env=env)
+
```
Comment (alex-voodoo) [inline testing/xvfb.py:82]: Perhaps we could decide here at run time whether we need to launch weston (depending on command line flags or environment) but I am not sure what to check.  In the other hand, weston seems to launch quite fast so perhaps the overhead from running it always isn't worth making the code more complex.

File: testing/xvfb.py [hunk 2]
```diff
@@ -75,13 +75,19 @@ def run_executable(cmd, env, stdoutfile=None):
         xcompmgr_proc = subprocess.Popen('xcompmgr', stdout=subprocess.PIPE,
                                          stderr=subprocess.STDOUT, env=env)
 
+        # Tests may need a Wayland server.
+        weston_proc = subprocess.Popen(('weston', '--backend=x11-backend.so'),
```
Comment (msisov) [inline testing/xvfb.py:79]: why --backend=x11-backend.so?

Comment (alex-voodoo) [reply testing/xvfb.py:79]: Because it couples naturally with X environment provided by xcompmgr and openbox (both are launched earlier).

**Comment by msisov** [comment]
after offline talk, I think it's lgtm

**Pull request merged**